pipelines/nlu-extension-build.yml

```diff
@@ -0,0 +1,74 @@
+trigger:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - /extensions/
+    - /pipelines/extension-build.yml
+pr:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - /extensions/
+    - /pipelines/extension-build.yml
+
+stages:
+- stage: Build_private
+  jobs:
+  - job: build_vsix
+    pool:
+      vmImage: 'ubuntu-latest' 
+    steps:
+    - task: NPM@1
+      displayName: npm install
+      inputs:
+        command: install
+        workingDir: extensions
+
+    - task: NPM@1
+      displayName: npm run lint
+      inputs:
+        command: custom
+        customCommand: run lint
+        workingDir: extensions
+
+    - task: NPM@1
+      displayName: npm run build
+      inputs:
+        command: custom
+        customCommand: run build
+        workingDir: extensions
+
+    - task: CopyFiles@2
+      displayName: Stage extension artifact for publish
+      inputs:
+        sourceFolder: extensions
+        contents: '**/*.vsix'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: PublishPipelineArtifact@0
```

Comment:
I don't think we need to publish the private pipeline artifact.

Reply:
If you decide to take the matrix / single pipeline YAML suggestion below, you could add a condition:
```
    - task: PublishPipelineArtifact@0
      condition: eq($(extensionVisibility), 'public')
      inputs:
        ...
```

Reply:
Added condition for pipeline artifact.